src/pyuvdata/uvdata/mwa_corr_fits.py

```diff
@@ -599,19 +607,29 @@ def flag_init(
             else np.reshape(self.flag_array, reshape)
         )
 
+        if freq_inds is not None:
+            nfreq_use = n_orig_freq
```

Comment:
should this check also look for `n_orig_freq` and not only `freq_inds`

Reply:
It could. The code that calls it should always pass in `n_orig_freq` if `freq_inds`
 is not None, but there's no enforcement of that in this function. I'll add that.